packages/@react-aria/tag/src/useTag.ts

```diff
@@ -43,58 +47,56 @@ export interface AriaTagProps<T> extends TagProps<T>, DOMProps, AriaLabelingProp
  * @param ref - A ref to a DOM element for the tag.
  */
 export function useTag<T>(props: AriaTagProps<T>, state: ListState<T>, ref: RefObject<FocusableElement>): TagAria {
-  let {
-    allowsRemoving,
-    item,
-    onRemove
-  } = props;
+  let {item} = props;
   let stringFormatter = useLocalizedStringFormatter(intlMessages);
-  let labelId = useId();
   let buttonId = useId();
 
-  let {rowProps, gridCellProps} = useGridListItem({
+  let {onRemove} = hookData.get(state) || {};
+  let {rowProps, gridCellProps, ...states} = useGridListItem({
     node: item
   }, state, ref);
 
   // We want the group to handle keyboard navigation between tags.
   delete rowProps.onKeyDownCapture;
+  delete states.descriptionProps;
 
   let onKeyDown = (e: KeyboardEvent) => {
     if (e.key === 'Delete' || e.key === 'Backspace') {
-      onRemove(item.key);
       e.preventDefault();
+      if (state.selectionManager.isSelected(item.key)) {
+        onRemove(new Set(state.selectionManager.selectedKeys));
+      } else {
+        onRemove(new Set([item.key]));
+      }
     }
   };
-  
+
   let modality: string = useInteractionModality();
   if (modality === 'virtual' &&  (typeof window !== 'undefined' && 'ontouchstart' in window)) {
     modality = 'touch';
   }
-  let description = allowsRemoving && (modality === 'keyboard' || modality === 'virtual') ? stringFormatter.format('removeDescription') : '';
+  let description = onRemove && (modality === 'keyboard' || modality === 'virtual') ? stringFormatter.format('removeDescription') : '';
   let descProps = useDescription(description);
 
-  let domProps = filterDOMProps(props);
```

Comment:
Removed this because it didn't actually do anything except put `id` on the grid cell, which is sort of a weird place for it? I guess data attributes too, but where do we want to put those?